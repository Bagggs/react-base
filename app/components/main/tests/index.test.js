import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Main from 'components/main';

describe('Main', () => {
  it('renders correctly', () => {
    const mainComponent = renderer.create(<Main/>).toJSON();

    expect(mainComponent).toMatchSnapshot();
  });

  describe('when session is created', () => {
    it('renders Todo container', () => {
      const middlewares = [thunk];
      const mockStore = configureStore(middlewares);
      const store = mockStore({ todos: { todos: [] }});
      const mainComponent = renderer.create(
        <Provider store={ store }>
          <Main loggedIn={ true }/>
        </Provider>
      ).toJSON();

      expect(mainComponent).toMatchSnapshot();
    });
  });
});
