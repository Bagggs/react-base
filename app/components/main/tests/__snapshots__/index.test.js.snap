// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h1>
      React-base
    </h1>
    <p>
      Kick-start your new web application based on React and Redux technologies.
      <br />
      It also includes Webpack, React hot loader, PostCSS, JSON-server tools for even more rapid development.
    </p>
  </div>
</div>
`;

exports[`Main when session is created renders Todo container 1`] = `
<div
  className="container"
>
  <div
    className="show-grid row"
  >
    <div
      className="col-md-8"
    >
      <h2>
        Todo List
      </h2>
    </div>
    <div
      className="col-md-4"
    >
      <button
        className="btn btn-primary pull-right spacingTop btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        New Task
      </button>
    </div>
  </div>
  <div
    className="show-grid row"
  >
    <div
      className="col-md-6"
    >
      <h3
        className="spacing-bottom"
      >
        Incomplete
      </h3>
      <ul
        className="list-group"
      />
    </div>
    <div
      className="col-md-6"
    >
      <h3
        className="spacing-bottom"
      >
        Complete
      </h3>
      <ul
        className="list-group"
      />
    </div>
  </div>
</div>
`;
